tests/usbus/main.c

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright (C) 2019 Koen Zandberg <koen@bergzand.net>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Tests USB interaction with USBUS
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include <errno.h>
+
+#include "mutex.h"
+#include "periph/usbdev.h"
+#include "embUnit.h"
+
+#include "usbdev_mock.h"
+#include "usb/descriptor.h"
+#include "usb/usbus.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/**
+ * @brief Test case states
+ */
+typedef enum {
+    TESTPHASE_START,        /**< Initial start */
+    TESTPHASE_RESET,        /**< Signal USB RESET */
+    TESTPHASE_REQ_DEV_INIT, /**< Request the first part of the device descriptor */
+    TESTPHASE_RESET2,       /**< Second reset */
+    TESTPHASE_SET_ADDRESS,  /**< Configure Address */
+    TESTPHASE_REQ_DEV_FULL, /**< Request the full device descriptor */
+    TESTPHASE_FINAL,        /**< Testing done */
+} testphase_t;
+
+/**
+ * @brief Test control request phases
+ */
+typedef enum {
+    TEST_REQ_PHASE_IDLE,
+    TEST_REQ_PHASE_START_GET,
+    TEST_REQ_PHASE_START_SET,
+    TEST_REQ_PHASE_OUTDATA,
+    TEST_REQ_PHASE_INDATA,
+    TEST_REQ_PHASE_OUTACK,
+    TEST_REQ_PHASE_INACK,
+} test_req_phase_t;
+
+testphase_t phase = TESTPHASE_START;
+test_req_phase_t req_phase = TEST_REQ_PHASE_IDLE;
+
+static usbus_t usbus;
+static char _stack[USBUS_STACKSIZE];
+
+static mutex_t waiter = MUTEX_INIT;
+
+static char *_state2str(test_req_phase_t state)
+{
+    switch (state) {
+        case TEST_REQ_PHASE_IDLE:
+            return "idle";
+        case TEST_REQ_PHASE_START_GET:
+            return "start get";
+        case TEST_REQ_PHASE_START_SET:
+            return "start set";
+        case TEST_REQ_PHASE_OUTDATA:
+            return "OUT data";
+        case TEST_REQ_PHASE_INDATA:
+            return "IN data";
+        case TEST_REQ_PHASE_OUTACK:
+            return "OUT Ack";
+        case TEST_REQ_PHASE_INACK:
+            return "IN Ack";
```

Comment:
missing default statement

Reply:
Intentionally missing, in that case the empty string at [line 87](https://github.com/RIOT-OS/RIOT/pull/12267/files/a33c46f3f7683ad6b8e93bdf954b942da5dffed8#diff-e989200794ef37dacd03b000cad21afcR87) will be returned

Reply:
I can change the empty string to something like `"unknown control request state"` though.

Reply:
Fixed